sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java

```diff
@@ -0,0 +1,230 @@
+package org.sdase.commons.server.mongo.testing;
+
+import static de.flapdoodle.embed.mongo.distribution.Version.Main.V3_6;
+import static de.flapdoodle.embed.mongo.distribution.Version.Main.V4_0;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import de.flapdoodle.embed.mongo.distribution.IFeatureAwareVersion;
+import de.flapdoodle.embed.mongo.distribution.Version;
+import org.apache.commons.lang3.SystemUtils;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * JUnit Test rule for running a MongoDB instance alongside the (integration) tests. Can be
+ * configured with custom user credentials and database name. Use {@link #getHosts()} to retrieve
+ * the hosts to connect to.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ * &#64;ClassRule
+ * public static final MongoDbRule RULE = MongoDbRule
+ *     .builder()
+ *     .withDatabase("my_db")
+ *     .withUsername("my_user")
+ *     .withPassword("my_s3cr3t")
+ *     .build();
+ * </pre>
+ */
+public interface MongoDb {
+
+  /**
+   * {@value} is the name of the environment variable that may hold a <a
+   * href="https://docs.mongodb.com/manual/reference/connection-string/">MongoDB Connection
+   * String</a> of the database used in tests instead of starting a dedicated instance.
+   */
+  String OVERRIDE_MONGODB_CONNECTION_STRING_ENV_NAME = "TEST_MONGODB_CONNECTION_STRING";
+
+  /**
+   * @return the hostname and port that can be used to connect to the database. The result may
+   *     contain a comma separated list of hosts as in the MongoDB Connection String
+   * @deprecated in favor of {@link #getHosts()}
+   */
+  @Deprecated
+  default String getHost() {
+    return getHosts();
+  }
```

Comment:
As this is deprecated, we shouldn't provide it in `MongoDbClassExtension`

Reply:
Absolutely correct. I updated the code accordingly.